### PR TITLE
Remove unused cliff gap parameters

### LINKED ISSUE #5341: Remove unused cliff gap parameters
This is necessary due to changes made in https://github.com/PolicyEngine/policyengine-us/pull/5322

### PR DESCRIPTION
**Pull request by anth-volk**
Fixes #5341.

This removes cliff gap parameters that are now unused (and therefore confusing to the package user) as a result of https://github.com/PolicyEngine/policyengine-us/pull/5322.

### CODE REVIEW

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: We would need to remove `cliff_evaluated` - the variable and the tests

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    removed:
+    - Unused cliff impact parameters
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Unused cliff impact parameters.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (anth-volk) [reply changelog_entry.yaml:1]: Can't believe I overlooked that, thank you. Will work on and re-request review when ready.

Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: Also `cliff_gap`.py and .yaml

Comment (anth-volk) [reply changelog_entry.yaml:1]: In this new round, I've modified `cliff_evaluated` to use the proper param and added a test to demonstrate the difference between how adults and children are handled in a given household for the variable (this threw me for a loop for a bit)

Comment (anth-volk) [reply changelog_entry.yaml:1]: Does `cliff_gap.py` refer to `cliff.py`'s definition of the `cliff_gap` variable?

Comment (anth-volk) [reply changelog_entry.yaml:1]: If so, I'm not sure I understand the modification needed there

Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: my mistake, I just meant the `cliff_gap` class need adjustments as well as the corresponding test

Comment (anth-volk) [reply changelog_entry.yaml:1]: Could you describe the adjustments? I have updated `cliff_gap`'s tests to work correctly, but am unsure on what's needed on the var itself.

Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: if it passes then it LGTM

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/household/cliff.py [hunk 2]
```diff
@@ -10,9 +10,15 @@ class cliff_evaluated(Variable):
 
     def formula(person, period, parameters):
         adult_index_values = person("adult_index", period)
-        cliff_adult_count = parameters(period).simulation.cliff_adults
+        mtr_adult_count = parameters(
+            period
+        ).simulation.marginal_tax_rate_adults
         is_adult = person("is_adult", period)
-        return is_adult & (adult_index_values <= cliff_adult_count)
+        with open("test_file.txt", "a") as f:
```
Comment (MaxGhenis) [inline policyengine_us/variables/household/cliff.py:17]: can we remove this?

Comment (anth-volk) [reply policyengine_us/variables/household/cliff.py:17]: Ugh, I hadn't meant to push that. Thanks for checking, let me look over all the changes with fresh eyes and get this merge-ready.

**Comment by anth-volk** [comment]
This should now be good to go

**Pull request merged**